config/template.js

```diff
@@ -56,27 +57,21 @@ const TEMPLATE_MAINTAINER = Joi
 
 const TEMPLATE_IMAGES = Joi.object()
     .pattern(Regex.IMAGE_ALIAS, Job.image)
-    .min(1)
-    .options({
-        language: {
-            object: {
-                allowUnknown: 'only supports the following characters A-Z,a-z,0-9,-,_'
-            }
-        }
-    });
+    .messages({
+        'object.unknown': '{{#label}} only supports the following characters A-Z,a-z,0-9,-,_'
+    })
+    .min(1);
 
 const SCHEMA_TEMPLATE = Joi.object()
     .keys({
         namespace: TEMPLATE_NAMESPACE,
-        name: TEMPLATE_NAME,
-        version: TEMPLATE_VERSION,
-        description: TEMPLATE_DESCRIPTION,
-        maintainer: TEMPLATE_MAINTAINER,
-        config: Job.job,
+        name: TEMPLATE_NAME.required(),
+        version: TEMPLATE_VERSION.required(),
+        description: TEMPLATE_DESCRIPTION.required(),
+        maintainer: TEMPLATE_MAINTAINER.required(),
+        config: Job.job.required(),
         images: TEMPLATE_IMAGES
-    })
-    .requiredKeys('name', 'version', 'description', 'maintainer',
-        'config', 'config.image', 'config.steps');
```

Comment:
is there any way to account for the requires `config.image` and `config.steps`?

Reply:
Added the check for `config.image` and `config.steps`. 
Although, there is hardly any documentation for requiredKeys alternative in hapi 17, we may need to update some base schemas in some cases.